diff --git a/h3-quinn/src/lib.rs b/h3-quinn/src/lib.rs
index c952a80..85d869e 100644
--- a/h3-quinn/src/lib.rs
+++ b/h3-quinn/src/lib.rs
@@ -25,8 +25,6 @@ use h3::{
     error::Code,
     quic::{self, ConnectionErrorIncoming, StreamErrorIncoming, StreamId, WriteBuf},
 };
-use tokio_util::sync::ReusableBoxFuture;
-
 #[cfg(feature = "tracing")]
 use tracing::instrument;

@@ -362,21 +360,11 @@ where
 ///
 /// Implements a [`quic::RecvStream`] backed by a [`quinn::RecvStream`].
 pub struct RecvStream {
-    stream: Option<quinn::RecvStream>,
-    read_chunk_fut: ReadChunkFuture,
+    stream: quinn::RecvStream,
     id: StreamId,
     is_0rtt: bool,
-    pending_stop: Option<VarInt>,
 }

-type ReadChunkFuture = ReusableBoxFuture<
-    'static,
-    (
-        quinn::RecvStream,
-        Result<Option<quinn::Chunk>, quinn::ReadError>,
-    ),
->;
-
 impl RecvStream {
     fn new(stream: quinn::RecvStream) -> Self {
         let id = u64::from(stream.id())
@@ -384,12 +372,9 @@ impl RecvStream {
             .expect("invalid stream id");
         let is_0rtt = stream.is_0rtt();
         Self {
-            stream: Some(stream),
-            // Should only allocate once the first time it's used
-            read_chunk_fut: ReusableBoxFuture::new(async { unreachable!() }),
+            stream,
             id,
             is_0rtt,
-            pending_stop: None,
         }
     }
 }
@@ -402,18 +387,9 @@ impl quic::RecvStream for RecvStream {
         &mut self,
         cx: &mut task::Context<'_>,
     ) -> Poll<Result<Option<Self::Buf>, StreamErrorIncoming>> {
-        if let Some(mut stream) = self.stream.take() {
-            self.read_chunk_fut.set(async move {
-                let chunk = stream.read_chunk(usize::MAX, true).await;
-                (stream, chunk)
-            })
-        };
-
-        let (mut stream, chunk) = ready!(self.read_chunk_fut.poll(cx));
-        if let Some(error_code) = self.pending_stop.take() {
-            let _ = stream.stop(error_code);
-        }
-        self.stream = Some(stream);
+        let read = self.stream.read_chunk(usize::MAX, true);
+        let mut read = std::pin::pin!(read);
+        let chunk = ready!(read.as_mut().poll(cx));
         Poll::Ready(Ok(chunk
             .map_err(convert_read_error_to_stream_error)?
             .map(|c| c.bytes)))
@@ -422,11 +398,7 @@ impl quic::RecvStream for RecvStream {
     #[cfg_attr(feature = "tracing", instrument(skip_all, level = "trace"))]
     fn stop_sending(&mut self, error_code: u64) {
         let error_code = VarInt::from_u64(error_code).expect("invalid error_code");
-        if let Some(stream) = self.stream.as_mut() {
-            let _ = stream.stop(error_code);
-        } else {
-            self.pending_stop = Some(error_code);
-        }
+        let _ = self.stream.stop(error_code);
     }

     #[cfg_attr(feature = "tracing", instrument(skip_all, level = "trace"))]
